consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftRound.java

```diff
@@ -142,23 +142,28 @@ private void updateStateWithProposalAndTransmit(
         roundChanges,
         prepares);
     updateStateWithProposedBlock(proposal);
+    sendPrepare(block);
   }
 
   public void handleProposalMessage(final Proposal msg) {
     LOG.debug("Received a proposal message. round={}", roundState.getRoundIdentifier());
     final Block block = msg.getSignedPayload().getPayload().getProposedBlock();
 
     if (updateStateWithProposedBlock(msg)) {
-      LOG.debug("Sending prepare message. round={}", roundState.getRoundIdentifier());
-      try {
-        final Prepare localPrepareMessage =
-            messageFactory.createPrepare(getRoundIdentifier(), block.getHash());
-        peerIsPrepared(localPrepareMessage);
-        transmitter.multicastPrepare(
-            localPrepareMessage.getRoundIdentifier(), localPrepareMessage.getDigest());
-      } catch (final SecurityModuleException e) {
-        LOG.warn("Failed to create a signed Prepare; {}", e.getMessage());
-      }
+      sendPrepare(block);
+    }
+  }
+
+  private void sendPrepare(final Block block) {
```

Comment:
sure we've got tests for this - but confirming, COULD this push us over  "prepared/committed" boundaries (in the case of 1 or 2 validator systems)? And if so - do we appear to operate sensibly?

Reply:
Yes we have test for this in QbftRoundTest singleValidatorImportBlocksImmediatelyOnProposalCreation that tests this